### PR TITLE
Implementing native type-hints

### PR DESCRIPTION
**Pull request by boesing**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | yes
| New Feature   | no

### Description

This patch provides native type-hints to any class' properties and methods.

### CODE REVIEW

**Comment by boesing** [comment]
Psalm phpunit plugin does not yet support v10 attributes.
I do not have time to actually implement it and thus, putting those errors to the baseline for now.

File: psalm.xml.dist [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        <directory name="src"/>
+        <directory name="test"/>
+        <ignoreFiles>
+            <directory name="vendor"/>
+            <directory name="test/TestAsset"/>
+            <directory name="test/**/TestAsset"/>
+        </ignoreFiles>
+    </projectFiles>
+
+    <issueHandlers>
```
Comment (Ocramius) [inline psalm.xml.dist:26]: I'd let these be baselined

**Review by Ocramius — APPROVED** [review]
LGTM: good improvement.

Merge at your own discretion.

File: psalm.xml.dist (hunk 1 above)
Comment (boesing) [reply psalm.xml.dist:26]: I already posted my opinion about that. psalm will remove this issue completely in one of the next majors as mixed is a valid value and unless it is a valid Argument somewhere, having these baselined or suppressed provides 0 value.

**Pull request merged**